File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
import java.util.*;

class Solution {
    public String addSpaces(String s, int[] spaces) {
        int n = s.length();
        int m = spaces.length;
        StringBuilder res = new StringBuilder();
        int j = 0;

        for (int i = 0; i < n; i++) {
            if (j < m && i == spaces[j]) {
                res.append(" ");
                j++;
            }
            res.append(s.charAt(i));
        }

        return res.toString();
    }
}

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int n = s.length();
        int m = spaces.size();
        string res ="";
        for(int i=0,j=0;i<n;i++){
            if(j<m and i == spaces[j]){
                res+=" ",j++;
            }
            res+=s[i];
        }
        return res;
    }
};